openwillis/measures/video/eye_blink.py

```diff
@@ -0,0 +1,285 @@
+        leftEye = np.array([(face_landmarks.landmark[lidx].x, face_landmarks.landmark[lidx].y) for lidx in LEFT_EYE_INDICES], dtype=np.float32)
+        rightEye = np.array([(face_landmarks.landmark[ridx].x, face_landmarks.landmark[ridx].y) for ridx in RIGHT_EYE_INDICES], dtype=np.float32)
+        return leftEye, rightEye
+    return None, None
+
+
+def detect_blinks(framewise, prominence, width):
+    """
+    ---------------------------------------------------------------------------------------------------
+
+    This function detects the blinks from a given EAR array.
+
+    Parameters:
+    ............
+    framewise : pd.DataFrame
+        Contains the frame number and the eye aspect ratio (EAR) of each frame
+    prominence : float
+        The prominence of the peaks
+    width : float
+        The width of the peaks
+
+    Returns:
+    ............
+    troughs : array
+        Array containing the frame number of each blink
+    left_ips : array
+        Array containing the frame number of the start of each blink
+    right_ips : array
+        Array containing the frame number of the end of each blink
+
+    ---------------------------------------------------------------------------------------------------
+    """
+    troughs, properties = find_peaks(-framewise['EAR'], prominence=prominence, width=width)
+    left_ips = properties["left_ips"]
+    right_ips = properties["right_ips"]
+    # round to nearest integer and add 1 to match frame number
+    left_ips = np.round(left_ips).astype(int) + 1
+    right_ips = np.round(right_ips).astype(int) + 1
+    troughs = np.round(troughs).astype(int) + 1
+    return troughs, left_ips, right_ips
+
+
+def convert_frame_to_time(troughs, left_ips, right_ips, fps):
+    """
+    ---------------------------------------------------------------------------------------------------
+
+    This function converts the frame number to time.
+
+    Parameters:
+    ............
+    troughs : array
+        Array containing the frame number of each blink
+    left_ips : array
+        Array containing the frame number of the start of each blink
+    right_ips : array
+        Array containing the frame number of the end of each blink
+    fps : float
+        The fps of the video
+
+    Returns:
+    ............
+    blinks : pd.DataFrame
+        Contains for each blink the frame number and the time (in seconds)
+         start of the blink and end of the blink
+
+    ---------------------------------------------------------------------------------------------------
+    """
+    troughs_time = troughs/fps
+    left_ips_time = left_ips/fps
+    right_ips_time = right_ips/fps
+
+    blinks = pd.DataFrame({'Blink Peak Frame': troughs, 'Blink Starting Frame': left_ips, 'Blink Ending Frame': right_ips,
+                            'Blink Peak Time': troughs_time, 'Blink Starting Time': left_ips_time, 'Blink Ending Time': right_ips_time})
+    return blinks
+
+
+def eye_blink_rate(video_directory):
```

Comment:
Just a suggestion: split this functions into smaller function for better readability

Reply:
[move framewise calculation to separate function](https://github.com/bklynhlth/openwillis/pull/21/commits/19090b5b12af2d50e92ba1d51d32476b7500ca36)